app/models/transaction.rb

```diff
@@ -148,15 +155,30 @@ def set_date
     self.date = Time.current
   end
 
+  def calculate_amount
+    self.amount = Money.new(amount_cents - comission.to_d * 100, currency)
+    add_comission_to_comment
+  end
+
   def recalculate_amount
     bank_account.recalculate_amount!
     nil
   end
 
+  def add_comission_to_comment
+    self.comment += "\nComission: " +
+      humanized_money_with_symbol(Money.new(comission.to_d * 100, bank_account.currency),
+      symbol_after_without_space: true)
+  end
+
   def residue?
     self.transaction_type == 'Residue'
   end
 
+  def update_invoice_paid_at
+    self.invoice.update(paid_at: self.date)
```

Comment:
@yunixon need to make sure that we will not update Invoice in other organization. Please add conditions to this query, so we will search Invoice only in Transaction Organization.